File: packages/typia/src/transformers/features/json/JsonIsParseTransformer.ts
import { JsonIsParseProgrammer } from "../../../programmers/json/JsonIsParseProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonIsParseTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "json.isParse",
      write: JsonIsParseProgrammer.write,
    });
}

File: packages/typia/native/transform/features/json/JsonIsParseTransformer.go
package json

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativejsonprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/json"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type jsonIsParseTransformerNamespace struct{}

var JsonIsParseTransformer = jsonIsParseTransformerNamespace{}

func (jsonIsParseTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "json.isParse",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativejsonprogrammers.JsonIsParseProgrammer.Write(x)
    },
  })
}
